restli-client/src/main/java/com/linkedin/restli/client/ParSeqBasedCompletionStage.java

```diff
@@ -561,10 +561,27 @@
     return _task.toCompletionStage().toCompletableFuture();
   }
 
-  /* ------------- For testing -------------- */
-
-  Task<T> getTask()
+  protected Task<T> getTask()
   {
     return _task;
   }
+
+  /**
+   * Special treatment to ParSeqBasedCompletionStage
+   * Note that there is no assumption that generated Task has been started
+   * The code which uses this method should consider that
+   */
+  @SuppressWarnings({"unchecked"})
+  protected <U> Task<U> getOrGenerateTaskFromStage(CompletionStage<? extends U> stage)
```

Comment:
Just a 2$, maybe I am just being naive.
Here is what I think, can we give flexibility to user in terms of what they want in the condition of the `else` part?
I am assuming this function here is act very similar to what we usually see in other frameworks: `getOrElse`
Just as this example https://medium.com/@Shehaaz/scala-options-explained-11d25fd67268 

Just to verify my understanding here, this `getOrGenerateTaskFromStage`  functionis trying to get a `Task` from a `CompletionStage`, but if it is not Parseq CompletionStage, then we generate a `Task` from whatever stage it is.

So I was thinking maybe we could let the caller provide a default Task?

Reply:
We can name it like `getTaskOrElse`, and the function argument can take a default `Task` from user's input?

Reply:
no, as the name suggest, "getOrGenerate", there is no "else" (since else means generate).